### PR TITLE
SAK-43580 - Site Info > Manage Groups > Create Group > no longer able…

### PR DESCRIPTION
**Pull request by mpellicer**
… to assign rosters

### CODE REVIEW

File: site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/GroupController.java [hunk 1]
```diff
@@ -290,8 +307,17 @@ public String saveGroup(@ModelAttribute GroupForm groupForm, Model model) {
                         addedGroupMemberList.add(member.getUserId());
                     }
                 }
-            // If the selected member is not a role, add it as individual member.
+            } else if (sectionList.stream().anyMatch(s -> selectedGroupMember.equals(s.getTitle()))) {
+                // If the selected member is a section, add all the members of this section to the group
+                selectedProviderIdList.add(selectedGroupMember);
+                Optional<Group> sectionOptional = sectionList.stream().filter(s -> selectedGroupMember.equals(s.getTitle())).findAny();
+                Group section = sectionOptional.get();
```
Comment (ern) [inline site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/GroupController.java:314]: The use of Optional.get() can produce a NoSuchElementException so it's best not to use this and use other variants of Optional, see https://www.baeldung.com/java-optional

Comment (mpellicer) [reply site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/GroupController.java:314]: I know but there is an IF statement before that checks that

Comment (ern) [reply site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/GroupController.java:314]: see ifPresent or isPresent no need to call it twice

Comment (mpellicer) [reply site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/GroupController.java]: good, replaced the logic using orElse

**Pull request merged**